src/components/my-page/contents/daily-check/CheckIn.tsx

```diff
@@ -1,9 +1,86 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import DayCheckBox from './DayCheckBox';
+import EmojiKeyboard from './EmojiKeyboard';
 import * as S from './style';
+import USER from '@/app/api/user';
+import { UserAttendanceDataType } from '@/types/user';
 
 export default function CheckIn() {
+  const [attendanceData, setAttendanceData] = useState<UserAttendanceDataType>({
+    no: 0,
+    nickname: '',
+    attendance: {},
+  });
+
+  const days: Record<number, string> = {
+    0: '월',
+    1: '화',
+    2: '수',
+    3: '목',
+    4: '금',
+    5: '토',
+    6: '일',
+  };
```

Comment:
@chamjin  이렇게 객체의 key를 `0부터 시작하는 숫자`로 다룰거면 배열로 관리하는게 어떨까요?

```suggestion
const days = ['월', '화', '수', '목', '금', '토', '일'];
```

```ts
<S.ContentSection>
  {days.map((day, index) => {
    const attendance = attendanceData.attendance && attendanceData.attendance[index] ? attendanceData.attendance[index] : [0, 0]

    return (
      <DayCheckBox
        key={day}
        day={day} // 요일 이름
        attendance={attendance} // 출석 정보
      />
    )
  })}
</S.ContentSection>
```
- 이제 기존처럼 `Object.keys`로 객체를 key만 모은 배열로 컨버팅 할 필요가 없어집니다.
- `Number()`로 key를 숫자로 변경 할 필요가 없습니다. 단순하게 index로 접근하시면 됩니다.
- 제가 혹시라도 놓친 부분이 있으면 말씀해주세요 :)

Reply:
오! 감사합니다
배열을 생각을 못하고 있었어요..
고치니까 코드가 더 깔끔해져서 좋네요 감사합니다!